test/unit/specs/addressbook.spec.js

```diff
@@ -217,4 +280,15 @@ describe("Addressbook", () => {
       .then(done.fail)
       .catch(() => done())
   })
+
+  it("should flag nodes that are not indexing transactions", async done => {
```

Comment:
as discussed, I think this is not the right way to go. the flag could change after we have discovered the peer the first time.